### DISCUSSION TITLE
How to specify index type?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2479] [Q&A] [answered]
I know that in an entity, I can create an index using `@Index({properties: ['machine']})` (I can provide a custom index name, but that is not necessery in my use case).

For example:

```sql
CREATE INDEX machines_id_idx ON md.machines USING btree (machine);
```

How can I specify to use `btree`? And generally, how does MikroORM decide what index type to use?

**Comment by B4nan** [comment, marked as answer]
With v5 you can use index expressions to have absolute control over the index.

```ts
@Index({ expression: 'CREATE INDEX machines_id_idx ON md.machines USING btree (machine);', name: 'machines_id_idx' })
```

For this it might be enough to use `@Index({ type: '...' })`, it control the `type` parameter of knex's `table.index()`:

https://github.com/mikro-orm/mikro-orm/blob/3bdb15d9bc31b8820c7891ffe2e451c3a23c7ca1/packages/knex/src/schema/SchemaGenerator.ts#L413

(for more info about that search knex docs and issues, I am not fully sure how it works)
